Add AOMP_BUILD_SYSTEM to change cmake generator

pdhaliwal-amd opened the pull request:
This adds support for configuring different build system than make.
Usage:
```
$ export AOMP_BUILD_SYSTEM=Ninja # value should be same as the name of any cmake generator
$ ./build_aomp.sh
```

For now, I have made changes to build_project.sh only as that is most time consuming part of the build. By default, make will be used for building.

ronlieb reviewed (changes requested):
do we also need for the make command to use ninja when this is enabled?

pdhaliwal-amd commented:
No, by default it should fallback to the old behaviour which was to use make.

estewart08 committed:
Change to aomp-dev branches. (#197)

Default aomp branch would be aomp-dev. Users can checkouy release branches for
source builds.

estewart08 committed:
Revert "Use amd-master branch of roctracer/rocprof with locked SHA."

This reverts commit 9e990a3e77a60e76c63c6861d462e9739c216ad8.

ronlieb committed:
Add map struct problem targc-272328

ronlieb committed:
put run_epsdb into aomp-dev branch, so aomp_clone_test will succeed

ronlieb committed:
Add nested declare target test

ronlieb committed:
Add ancestorC test for reverse target offload ancestor clause

ronlieb committed:
Add fortran variant of anecstor clause

ronlieb committed:
Add icvars-nested  test

ronlieb committed:
Add icvar-nest fortran test

ronlieb committed:
Add canonical loop !=

JonChesterfield committed:
Add test runner for the linkable openmp examples

estewart08 committed:
[Test] - loop_order added to smoke fails.

ronlieb committed:
Add collapse nonrectangular test

ronlieb committed:
Add flang collapse nonrect test

estewart08 committed:
Add declare_target_pointer from AOMP issue #198

ronlieb committed:
Add pragma omp loop

ronlieb committed:
Add reduction_array test

estewart08 committed:
Remove cleanup from check_omp5.sh, loop_order now correctly returns 0 if successful.

estewart08 committed:
[Test] - add flang_atomic_hint to smoke-fails.

ronlieb committed:
Remove sha-key for libdevice for aomp12, need latest amd-stg-open

estewart08 committed:
Add run all script. (#200)

* Add run all script.

Still needs some work/improvement.
Raja, omptests are disabled for now as they are crashing my GPU.
QMCPACK will timeout after 20mins. This is done to prevent the soft hang after the
compilation fails. Normally this build step would take about 17 minutes.

Total run time is around an hour.

* Move export NUM_THREADS after aomp_common_vars is sourced. This speeds up build times on various suites.

* Remove username from script.

estewart08 committed:
Do not use space padded day of the month as it messes up log names.

ronlieb committed:
make evars for SUITE_LIST and EPSDB_LIST

estewart08 committed:
Update run_test_suite.sh and various test scripts.

run_test_suite.sh
- Added --foreground to timeout for qmcpack to kill process faster if user cancels.
- Added smokefails to default list.
- Adjust smoke/smokefails/omp5 to use the new gatherdata function for reporting results.

check_smoke.sh/check_omp5.sh/check_smoke_fails.sh
- Moved printing of results to gatherdata function.
- Function can be used by other scripts via 'script_name.sh gatherdata'.
- Allows easier reporting for the run_test_suite.sh

estewart08 committed:
[DOCS] - Replace aomp11 references with aomp13.

gregrodgers committed:
We are now forcing the need to specify library name if fat library contains device code that must be unbundled

gregrodgers committed:
Merge branch 'aomp-dev' of https://github.com/ROCm-Developer-Tools/aomp into aomp-dev

gregrodgers committed:
Fix override of QMCPACK_REPO. Allow 'nocmake' option to restart previous build of qmcpack.  Be specific about locations of LLVM compilers and mpicc and mpicxx

gregrodgers committed:
ROCm 4.1 appends .kd on kernel names when querying symbols

gregrodgers committed:
removed hardcoding of /opt/rocm to use

gregrodgers committed:
remove hardcoding of /opt/rocm

gregrodgers committed:
removed hardcoding of /opt/rocm.  The EXTRA_LDFLAGS now use -L/lib and AOMPROCM is set to /tmp/grodgers/rocm/aomp if not already set.

gregrodgers committed:
update developers readme to describe aomp13 development.  It was aomp11

gregrodgers committed:
More cleanup of developer readme

gregrodgers committed:
More edits and fixes.  New table for important environment variables

gregrodgers committed:
md does not support merging rows in a table

gregrodgers committed:
Change default alternative linker to AOCC 3.0.0

estewart08 committed:
aomp_common_vars - Lock comgr sha for aomp13.

estewart08 committed:
smoke_fails - Add exit(1) to show tests are failing. These tests were incorrectly giving a passing return code

estewart08 committed:
complex-debug - move from smoke-fails to smoke as test is passing with 13.0-2.

estewart08 committed:
Smoke Makefile - Add missing tests

estewart08 committed:
gdb_teams use AOMPROCM instead of AOMP in run recipe.

ronlieb committed:
Add simple a+a test

ronlieb committed:
Cleanup A+Atest

ronlieb committed:
dont set path

estewart08 committed:
Bump version to 13.0-3. Release branches for 13.0-2 were created.

ronlieb committed:
Add -v toa+atest

gregrodgers committed:
remove sha key for rocm-device-libs and rocm-compilersupport because they are not needed in aomp-13.0-3

estewart08 committed:
Merge aomp 13.0-2 to aomp dev (#201)

* aomp_common_vars - use 13.0-2 release branches

* aomp_common_vars - Remove unnecessary echos

* Fixes to aomp_common_vars and debian/rules.

aomp_common_vars
  -cd to roctracer was causing issues with scripts sourcing this file, change to pushd/popd
debian/rules
  -add new library path for librocprofiler64.so to prevent the debian build from failing

* Add back previous patch for comgr only on 16.04. This gets past the comgr build failure.

* Update docs, README, and changelog for 13.0-2 release.

* Remove c++17 from Kokkos example Makefile

* Adjust Docs, add math_pow to smoke Makefile.

* Update docs, README, and source install methods.

spack package.py
  -suppress flang/flang_runtime warnings, which spack thought were errors.
  -update amd-llvm-project, llvm-project
tarball install
  -remove roctracer/rocprofiler from component list as these are not in the tarball
  -add TARBALL_INSTALL=1 to Makefile
update changelog
remove references to aomp11 in patch README

* Add TARBALL_INSTALL variable to build_aomp.sh for removal of roctracer/rocprofiler from component list.

* After release branch merge to aomp-dev. Change aomp_common_vars/docs to aomp-13-0-3 dev nomenclature.

gregrodgers committed:
Do not build tests for Ubuntu 16.04 because it requires python3.6 which Ubuntu 16.04 does not have

gregrodgers committed:
remove hardcoding of EXTRA_CFLAGS to blank so adding EXTRA_CFLAGS is possible without editing Makfiles.  For those tests that set EXTRA_CFLAGS, I just changed = to +=

raghavendhra committed:
Moved flang-272730-complex from smoke-fails/ to smoke/
	modified:   ../smoke-fails/Makefile
	modified:   Makefile
	renamed:    ../smoke-fails/flang-272730-complex/Makefile -> flang-272730-complex/Makefile
	renamed:    ../smoke-fails/flang-272730-complex/flang272730-complex.f90 -> flang-272730-complex/flang272730-complex.f90

ronlieb committed:
change llvm-project to use lld

JonChesterfield committed:
minor changes to run_test_suite

gregrodgers committed:
This is a port of the roctracer tester to AOMP.  Just run test/run.sh

gregrodgers committed:
Merge branch 'aomp-dev' of https://github.com/ROCm-Developer-Tools/aomp into aomp-dev

gregrodgers committed:
fix so run.sh can really be called from anywhere

ronlieb committed:
Add famd-opt-279975 test

ronlieb committed:
Move flang-275472 to smoke-fails

ronlieb committed:
two closed bugs

ronlieb committed:
Fix test targ-273742 to call omp_target_alloc with a device number rather than host

ronlieb committed:
Add more outputto hip_host_register test

ronlieb committed:
Add hipHostGetDevicePointer to test hip_host_register to get it to work

ronlieb committed:
Increase log output for run_epsdb

richardbleikamp committed:
add smoke-fail test for SOMD clause not supported yet.

ronlieb committed:
Add two alt tests t smoke-fails

ronlieb committed:
Add clang-274983-2

richardbleikamp committed:
iadd smoke-fails test for collapse(7) clause on omp parallel do directive

ronlieb committed:
modify flang-283928-collapse7 to compile for target

ronlieb committed:
update test to show use_device_ptr(x) works on device

ronlieb committed:
Add more tests for flang-275194

ronlieb committed:
remove race condition from 274775

ronlieb committed:
update 2 complex tests

estewart08 committed:
Update usage of AOMPHIP in hip_device_compile OmpHipMallocManged.

ronlieb committed:
Add flang-272343-2

gregrodgers committed:
add arg -DHSA_RUNTIME_LIB_PATH because updated cmake for roctracer needs it

pdhaliwal-amd committed:
Support building via ninja for llvm-project

ghost commented:
## DeepCode's analysis on [#aac8bd](https://www.deepcode.ai/app/gh/ROCm-Developer-Tools/aomp/00aaf49a9511d19fd86336e41d878a0b101ee8fd/ROCm-Developer-Tools/aomp/aac8bde7252bf8c87951dbe127eb2911b5319f1f/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=3&) found:

- :information_source: **3** minor issues. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Missing close for open, add close or use a with block. Occurrences: <ul><li><a href="https://github.com/ROCm-Developer-Tools/aomp/blob/aac8bde7252bf8c87951dbe127eb2911b5319f1f/examples/roctracer/test/check_trace.py#L35">check_trace.py:35</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/ROCm-Developer-Tools/aomp/00aaf49a9511d19fd86336e41d878a0b101ee8fd/ROCm-Developer-Tools/aomp/aac8bde7252bf8c87951dbe127eb2911b5319f1f/pr/_/%2Fexamples%2Froctracer%2Ftest%2Fcheck_trace.py/python%2Fdc%2FMissingClose%2Ftest/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Use the is operator instead of == to compare to None, because == calls __eq__ which is slower and error-prone. Occurrences: <ul><li><a href="https://github.com/ROCm-Developer-Tools/aomp/blob/aac8bde7252bf8c87951dbe127eb2911b5319f1f/examples/roctracer/test/check_trace.py#L40">check_trace.py:40</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/ROCm-Developer-Tools/aomp/00aaf49a9511d19fd86336e41d878a0b101ee8fd/ROCm-Developer-Tools/aomp/aac8bde7252bf8c87951dbe127eb2911b5319f1f/pr/_/%2Fexamples%2Froctracer%2Ftest%2Fcheck_trace.py/python%2Fdc%2Fchange_to_is%2Ftest/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">No need to call keys, directly check with the in operator. Occurrences: <ul><li><a href="https://github.com/ROCm-Developer-Tools/aomp/blob/aac8bde7252bf8c87951dbe127eb2911b5319f1f/examples/roctracer/test/check_trace.py#L156">check_trace.py:156</a></li> <li><a href="https://github.com/ROCm-Developer-Tools/aomp/blob/aac8bde7252bf8c87951dbe127eb2911b5319f1f/examples/roctracer/test/check_trace.py#L278">check_trace.py:278</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/ROCm-Developer-Tools/aomp/00aaf49a9511d19fd86336e41d878a0b101ee8fd/ROCm-Developer-Tools/aomp/aac8bde7252bf8c87951dbe127eb2911b5319f1f/pr/_/%2Fexamples%2Froctracer%2Ftest%2Fcheck_trace.py/python%2Fdc%2FUpdateAPI%2Ftest/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/ROCm-Developer-Tools/aomp/00aaf49a9511d19fd86336e41d878a0b101ee8fd/ROCm-Developer-Tools/aomp/aac8bde7252bf8c87951dbe127eb2911b5319f1f/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=3&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=ROCm-Developer-Tools)
